bindings/pydrake/BUILD.bazel

```diff
@@ -281,6 +278,19 @@ PY_LIBRARIES_WITH_INSTALL = [
     "//bindings/pydrake/visualization",
 ]
 
+# TODO(mmanu): Make this parallel the packages shown above (so that it's in one
```

Comment:
I'm sorry I don't know what you want me to do here.  Do you want me to use this inside `PY_LIBRARY_WITH_INSTALL`? 
`PY_LIBRARY_WITH_INSTALL = PYBIND_SUBPACKAGES_FOR_COVERAGE + [...]` ?

I can see a downside of not going the provider route.  This way, it seems to me, we're repeating ourselves.  It seems like we're giving at two places, which sub-packages we'd like coverage stats for.  Here and inside the build files, using the macro `add_pybind_coverage_data`.

If we went the provider route, we'd be telling Bazel only once, i.e. with the macro which packages we care about coverage for.

Reply:
I was thinking like this:
```
PY_LIBRARIES_AS_SUBPACKAGES = [ ... ]
PY_LIBRARIES_WITH_INSTALL = [...] + PY_LIBRARIES_AS_SUBPACKAGES
...
add_pybind_coverage_data(..., subpackages = PY_LIBRARIES_AS_SUBPACKAGES)
```

> If we went the provider route, we'd be telling Bazel only once, i.e. with the macro which packages we care about coverage for.

I don't understand how making a provider makes the targets discoverable by separate packages? I don't think you can query targets from a separate package easily?

Reply:
> I don't understand how making a provider makes the targets discoverable by separate packages?

[`provider` is the only way to share data](https://docs.bazel.build/versions/master/skylark/rules.html#providers) between rules, which are more fine than packages.  So, what we do is add a rule which fetches the data we want to share between packages and share it using providers.  We then accumulate that data using yet another rule and run our script on it.

But, my original argument was wrong, I realized that we would be indeed repeating ourselves even with providers.
https://github.com/RobotLocomotion/drake/pull/12040/files#diff-2eade61c44736139ca71679b970f81a4R529

Reply:
Yes, I knew the general bits about providers -- what I meant was "queryable", like what is provided via `native.existing_rules()` which is used in linting, etc, but from another package,  at analysis / loading time (forgot which).

And yeah, that's what your PR did before; my modification may not explicitly touch on providers, but it's effectively doing so by the fact that `native.filegroup` will use its own builtin provider:
https://github.com/m-chaturvedi/drake/blob/1a4d8763717240e7a46ce224adabce7799e493d9/tools/skylark/pybind.bzl#L428-L430